pkg/providers/vsphere/defaults.go

```diff
@@ -87,6 +91,18 @@ func (d *Defaulter) setDefaultTemplateIfMissing(ctx context.Context, spec *Spec,
 	return nil
 }
 
+func (d *Defaulter) setDefaultTemplateIfMissing(ctx context.Context, spec *Spec, m *anywherev1.VSphereMachineConfig) error {
+	if m.Spec.Template == "" {
+		logger.V(1).Info("VSphereMachineConfig template is not set. Using default template.")
+		versionsBundle := spec.ControlPlaneVersionsBundle()
```

Comment:
could we add a comment saying here that we assume we need to use the control plane versions bundle bc at this point all worker node machine configs should already have a template?

Or you can pass the versions bundle as an argument and move the comment up to `setDefaultsForMachineConfig` which is probably cleaner.

Reply:
We are going to address this separately